### PR TITLE
Added Orientations operations

### PR DESCRIPTION
**Pull request by jessemapel**
This adds inversion and multiplication to orientation objects. Note that the inversion is not perfect as the way we store them isn't full invertible. The frames do not fully invert and the constant rotation gets folded into the time dependent rotation and it becomes fully time dependent.

### CODE REVIEW

File: src/Orientations.cpp [hunk 1]
```diff
@@ -144,4 +150,60 @@ namespace ale {
...
+    for (size_t i = 0; i < m_rotations.size(); i++) {
+      newRotations.push_back(m_rotations[i].inverse() * constInverseRotation);
+    }
+
+    std::vector<Vec3d> rotatedAvs;
+    for (size_t i = 0; i < m_avs.size(); i++) {
+      Vec3d rotatedAv = (m_constRotation * m_rotations[i])(m_avs[i]);
+      rotatedAv.x *= -1.0;
+      rotatedAv.y *= -1.0;
+      rotatedAv.z *= -1.0;
```
Comment (jessemapel) [inline src/Orientations.cpp:170]: It may be worth it to just add a * operator to the vector struct.

File: include/ale/Orientations.h [hunk 2]
```diff
@@ -71,15 +71,27 @@ namespace ale {
...
      */
     Orientations &operator*=(const Orientations &rhs);
+
     /**
      * Multiply this orientation by a constant rotation
      */
     Orientations &operator*=(const Rotation &rhs);
 
+    /**
+     * Invert the orientations
```
Comment (jessemapel) [inline include/ale/Orientations.h:91]: Note for myself that I need to note all of the stuff that doesn't perfectly invert in here.

File: src/Orientations.cpp [hunk 3]
```diff
@@ -144,4 +150,60 @@ namespace ale {
...
+
+
+  Orientations operator*(Orientations lhs, const Rotation &rhs) {
+    return lhs *= rhs;
+  }
+
+
+  Orientations operator*(const Rotation &lhs, Orientations rhs) {
+    return rhs.addConstantRotation(lhs);
+  }
```
Comment (acpaquette) [inline src/Orientations.cpp:203]: This seems odd compared to the other two multiplication implementations.

Comment (jessemapel) [reply src/Orientations.cpp:203]: How so? There is a little strangeness about what's a reference and what's not. That's just trying to be specific about what needs to be copied and what doesn't.

**Pull request merged**